accounts-on-demand/src/subscription_manager.rs

```diff
@@ -0,0 +1,281 @@
+                                        filter: Some(Filter::Memcmp(
+                                            SubscribeRequestFilterAccountsFilterMemcmp {
+                                                offset: memcmp.offset,
+                                                data: Some(match &memcmp.data {
+                                                    MemcmpFilterData::Bytes(bytes) => {
+                                                        Data::Bytes(bytes.clone())
+                                                    }
+                                                    MemcmpFilterData::Base58(data) => {
+                                                        Data::Base58(data.clone())
+                                                    }
+                                                    MemcmpFilterData::Base64(data) => {
+                                                        Data::Base64(data.clone())
+                                                    }
+                                                }),
+                                            },
+                                        )),
+                                    }
+                                }
+                                AccountFilterType::TokenAccountState => {
+                                    SubscribeRequestFilterAccountsFilter {
+                                        filter: Some(Filter::TokenAccountState(false)),
+                                    }
+                                }
+                            })
+                            .collect_vec()
+                    } else {
+                        vec![]
+                    };
+                    subscribe_accounts.insert(
+                        format!("accounts_{}", program_id),
+                        SubscribeRequestFilterAccounts {
+                            account: vec![],
+                            owner: vec![program_id.clone()],
+                            filters,
+                        },
+                    );
+                }
+            }
+
+            let subscribe_request = SubscribeRequest {
+                accounts: subscribe_accounts,
+                slots: Default::default(),
+                transactions: Default::default(),
+                blocks: Default::default(),
+                blocks_meta: Default::default(),
+                entry: Default::default(),
+                commitment: Some(processed_commitment.into()),
+                accounts_data_slice: Default::default(),
+                ping: None,
+            };
+
+            log::info!(
+                "Accounts on demand subscribing to {}",
+                grpc_config.grpc_addr
+            );
+            let mut client = yellowstone_grpc_client::GeyserGrpcClient::connect(
+                grpc_config.grpc_addr.clone(),
+                grpc_config.grpc_x_token.clone(),
+                None,
+            )
+            .unwrap();
```

Comment:
**caution:** this will panic your thread

Reply:
if you consider using my auto connect it should be fine [huge autoconnect](https://github.com/blockworks-foundation/geyser-grpc-connector/blob/a6ab686fa4607e95550d29f77f08ea074b90e91f/src/grpc_subscription_autoreconnect_tasks.rs#L143)